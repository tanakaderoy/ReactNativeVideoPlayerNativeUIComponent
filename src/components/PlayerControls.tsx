import React from 'react';
import {StyleSheet, TouchableOpacity, View} from 'react-native';
import {Icon} from 'react-native-elements';
import {COLORS} from '../utils/Util';

interface Props {
  onSkipBack: () => void;
  onSkipForward: () => void;
  onPlayPause: () => void;
  isPlaying: boolean;
}

const PlayerControls = ({
  onSkipBack,
  onSkipForward,
  onPlayPause,
  isPlaying,
}: Props) => {
  const controlIconSize = 50;

  return (
    <View style={styles.playerControls}>
      <TouchableOpacity onPress={onSkipBack}>
        <Icon
          name="replay-5"
          color={COLORS.textColor}
          type="material"
          size={controlIconSize}
        />
      </TouchableOpacity>
      <TouchableOpacity onPress={onPlayPause}>
        <Icon
          name={
            isPlaying ? 'ios-pause-circle-outline' : 'ios-play-circle-sharp'
          }
          color={COLORS.textColor}
          type="ionicon"
          size={controlIconSize + 10}
        />
      </TouchableOpacity>

      <TouchableOpacity onPress={onSkipForward}>
        <Icon
          name="forward-5"
          color={COLORS.textColor}
          type="material"
          size={controlIconSize}
        />
      </TouchableOpacity>
    </View>
  );
};

export default PlayerControls;

const styles = StyleSheet.create({
  playerControls: {
    paddingHorizontal: 5,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-evenly',
    flex: 3,
  },
});
