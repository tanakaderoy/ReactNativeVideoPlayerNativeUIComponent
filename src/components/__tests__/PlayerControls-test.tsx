import React from 'react';
import * as renderer from 'react-test-renderer';
import PlayerControls from '../PlayerControls';
let isRan = false;
const func = () => {
  isRan = true;
};
it('renders player controls', () => {
  const playerControls = renderer
    .create(
      <PlayerControls
        onPlayPause={func}
        onSkipBack={func}
        onSkipForward={func}
        isPlaying={true}
      />,
    )
    .toJSON();
  expect(isRan).toEqual(false);
  expect(playerControls).toMatchSnapshot();
});

// /**
//  * @format
//  */

// import React from 'react';
// import 'react-native';
// // Note: test renderer must be required after react-native.
// import renderer, {act} from 'react-test-renderer';
// import App from '../App';

// it('renders correctly', () => {
//   act(() => {
//     renderer.create(<App />);
//   });
// });
