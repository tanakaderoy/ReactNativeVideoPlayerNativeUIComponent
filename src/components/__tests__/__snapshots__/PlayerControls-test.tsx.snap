// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders player controls 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 3,
      "flexDirection": "row",
      "justifyContent": "space-evenly",
      "paddingHorizontal": 5,
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "overflow": "hidden",
        }
      }
    >
      <View>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "justifyContent": "center",
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 50,
                },
                Object {
                  "backgroundColor": "transparent",
                },
                Object {
                  "fontFamily": "Material Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
            testID="iconIcon"
          >
            
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "overflow": "hidden",
        }
      }
    >
      <View>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "justifyContent": "center",
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 60,
                },
                Object {
                  "backgroundColor": "transparent",
                },
                Object {
                  "fontFamily": "Ionicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
            testID="iconIcon"
          >
            
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "overflow": "hidden",
        }
      }
    >
      <View>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "justifyContent": "center",
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 50,
                },
                Object {
                  "backgroundColor": "transparent",
                },
                Object {
                  "fontFamily": "Material Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
            testID="iconIcon"
          >
            
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`renders with defaults 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 3,
      "flexDirection": "row",
      "justifyContent": "space-evenly",
      "paddingHorizontal": 5,
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "overflow": "hidden",
        }
      }
    >
      <View>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "justifyContent": "center",
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 50,
                },
                Object {
                  "backgroundColor": "transparent",
                },
                Object {
                  "fontFamily": "Material Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
            testID="iconIcon"
          >
            
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "overflow": "hidden",
        }
      }
    >
      <View>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "justifyContent": "center",
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 60,
                },
                Object {
                  "backgroundColor": "transparent",
                },
                Object {
                  "fontFamily": "Ionicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
            testID="iconIcon"
          >
            
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "overflow": "hidden",
        }
      }
    >
      <View>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "justifyContent": "center",
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 50,
                },
                Object {
                  "backgroundColor": "transparent",
                },
                Object {
                  "fontFamily": "Material Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
            testID="iconIcon"
          >
            
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
